src/Project/functions/compileFiles.ts

```diff
@@ -142,7 +144,20 @@ export function compileFiles(
 
 				for (const sourceFile of transformResult.transformed) {
 					if (ts.isSourceFile(sourceFile)) {
-						updateFile(sourceFile.fileName, ts.createPrinter().printFile(sourceFile));
+						const source = ts.createPrinter().printFile(sourceFile);
+						updateFile(sourceFile.fileName, source);
+						if (data.projectOptions.writeTransformedFiles) {
+							let outPath = pathTranslator.getOutputPath(sourceFile.fileName);
+							outPath = outPath.substring(0, outPath.length - path.extname(outPath).length);
+							outPath += ".transformed.ts";
```

Comment:
It might be cleaner/safer to add a `pathTranslator.getTransformedOutputPath()` function which is similar to `getOutputPath`